Speedrun Timer Implementation

ChillBeanJosh opened the pull request:
-Added into JoshuaTestScene

-Implemented a StopWatch Timer into the UI that can be toggled on and off by a bool.

-Additionally, not sure if required created 2 test point objects called "StartArea" and "EndArea" which activate and deactivate the bool without having to use the inspector to test in real time.

(Note: None of the additions have been merged with any existing prefabs in the case that things need to be reworked)

["StopWatch Controller" attached to player]
["StopWatch" attached to TImerUI in Canvas]

ChillBeanJosh committed:
Speedrun Timer Implementation

-Added into JoshuaTestScene

-Implemented a StopWatch Timer into the UI that can be toggled on and off by a bool.

-Additionally, not sure if required created 2 test point objected called "StartArea" and "EndArea" which activate and deactivate the bool without having to use the inspector to test in real time.

(Note: None of the additions have been merged with any existing prefabs in the case that things need to be reworked)

TheStress commented:
Requested Changes:
- [ ] Turn the `TimerUI` into a prefab
- [ ] Instead of having the player check for the start and end areas have the areas check for the player instead (to not crowd the player prefab + separation of concern)
- [ ] Use the physics layer to check for the player instead of object name (the name is likely to change)
- [ ] Turn the start and end areas into prefabs too

Documentation:
- [ ] https://github.com/StudioAspen/AAAGame2023-2024/wiki/Writing-Documentation-For-Designers

(Note: Dont worry about conflicts I'll deal with them)

ChillBeanJosh committed:
Updated Timer

-Made the TimerUI a prefab inside the "Player" folder.

-Added the 2 Timer controlling objects as prefabs inside the "Placeable" folder.

-Changed to have the area to check for the player instead of player check for area.
(Using layer Mask, though me know if physics layer is something different)

-

TheStress committed:
updated prefabs + controllers find stopwatch

TheStress committed:
Merge branch 'develop' into Timer-for-Speedrunning
